packages/plugin-core/src/commands/GoToSiblingCommand.ts

```diff
@@ -33,65 +34,207 @@ export class GoToSiblingCommand extends BasicCommand<
+    if (!note) throw new Error(`${ctx}: ${UNKNOWN_ERROR_MSG}`);
+    // Get the sibling note
+    let siblingNote: NoteProps;
+    // If the active note is a journal note, get the sibling note based on the chronological order
+    if (await this.canBeHandledAsJournalNote(note, workspace.engine)) {
+      const { sibling, msg } = await this.getSiblingForJournalNote(
+        workspace.engine.notes,
+        note,
+        opts.direction
+      );
+      if (msg) return { msg };
+      siblingNote = sibling!;
+    } else {
+      const { sibling, msg } = this.getSibling(
+        workspace,
+        note,
+        opts.direction,
+        ctx
+      );
+      if (msg) return { msg };
+      siblingNote = sibling!;
+    }
+    // Get path to the vault
+    const vpath = vault2Path({
+      vault: siblingNote.vault,
+      wsRoot: workspace.wsRoot,
+    });
+    // Open the sibling note
+    await VSCodeUtils.openFileInEditor(
+      VSCodeUtils.joinPath(Uri.file(vpath), siblingNote.fname + ".md")
+    );
+    // Return the success message for testing
+    return { msg: "ok" as const };
+  }
 
-    const { vaults, engine, wsRoot } = ExtensionProvider.getDWorkspace();
-    if (value === "root") {
-      const vault = VaultUtils.getVaultByFilePath({
-        vaults,
-        wsRoot,
-        fsPath: maybeTextEditor.document.uri.fsPath,
-      });
-      const rootNode = NoteUtils.getNoteByFnameFromEngine({
-        fname: value,
-        vault,
-        engine,
-      });
-      if (_.isUndefined(rootNode)) {
-        throw new DendronError({ message: "no root node found" });
-      }
-      respNodes = rootNode.children
-        .map((ent) => engine.notes[ent])
-        .concat([rootNode]);
+  async getActiveNote(
+    engine: DEngineClient,
+    fname: string
+  ): Promise<NoteProps | null> {
+    const vault = PickerUtilsV2.getVaultForOpenEditor();
+    const hitNotes = await engine.findNotes({ fname, vault });
+    return hitNotes.length !== 0 ? hitNotes[0] : null;
+  }
+
+  private async canBeHandledAsJournalNote(
+    note: NoteProps,
+    engine: DEngineClient
+  ): Promise<boolean> {
+    const markedAsJournalNote =
+      NoteUtils.getNoteTraits(note).includes("journalNote");
+    if (!markedAsJournalNote) return false;
+
+    // Check the date format for journal note. Only when date format of journal notes is default,
+    // navigate chronologically
+    const config = await engine.getConfig();
+    const dateFormat = config.data?.workspace.journal.dateFormat;
+    return dateFormat === "y.MM.dd";
+  }
+
+  private async getSiblingForJournalNote(
+    allNotes: NotePropsByIdDict,
+    currNote: NoteProps,
+    direction: Direction
+  ): Promise<{ sibling: NoteProps | null; msg: CommandOutput["msg"] | null }> {
+    const journalNotes = this.getSiblingsForJournalNote(allNotes, currNote);
+    // If the active note is the only journal note in the workspace, there is no sibling
+    if (journalNotes.length === 1) {
```

Comment:
for user facing messages, we should try to show this in the outermost loop (inside the `execute` function)